### DISCUSSION TITLE
Welcome to netschoolapi Discussions!

### DISCUSSION THREAD

**Discussion by nm17** [nm17/netschoolapi#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TheEntropyShard** [comment]
Привет. Делаю свое апи но на джаве. Чтобы я не делал всегда получаю 409. Есть несколько вопросов:

- Зачем идет запрос к logindata?
- Зачем кодировка пароля изменяется на windows-1251?
- И параметры,  которые передаются в /webapi/login это ведь параметры адресной строки?

**Reply by lentryd** [reply]
Можешь посмотреть в этом репозитории https://github.com/lentryd/NetSchoolApi

**Reply by nm17** [reply]
1. Для получения данных о школах
2. Ибо пароли в браузере кодируются в windows 1251 и для русских паролей это обязательно
3. Вроде нет, post

**Reply by TheEntropyShard** [reply]
1. Но ведь logindata не возвращает ничего о школах
2. понял
3. понял

Все равно получаю {"message":"Ошибка входа в систему","details":null}. Вроде и какую то куку сохранил: NSSESSIONID, но не помогло. Сервер просто не хочет принимать мой запрос и все, и еще не понимаю одну вещь, сначала мы делаем запрос к getdata, а потом данные от туда используем для хэширования пароля и для логина, но как сервер сравнивает два хеша, если salt все время разный, сравнивает айпи с запросов к getdata и login?

**Reply by lentryd** [reply]
> 1. Но ведь logindata не возвращает ничего о школах

Вообще он возвращается ver и salt. ver просто как доп. параметр, который лучше использовать. salt используется для хеширования пароля.

**Reply by lentryd** [reply]
> не понимаю одну вещь, сначала мы делаем запрос к getdata, а потом данные от туда используем для хэширования пароля и для логина, но как сервер сравнивает два хеша, если salt все время разный...

Если мне не изменяет память, то salt не меняется

**Reply by TheEntropyShard** [reply]
ver salt и lt возвращает webapi/auth/getdata. И все они каждый запрос разные,

**Reply by lentryd** [reply]
> ver salt и lt возвращает webapi/auth/getdata. И все они каждый запрос разные, 

Так ты делаешь один запрос и потом просто передаешь эти данные. Вроде lt и используется для понимания сервером, какой salt использовать

**Reply by TheEntropyShard** [reply]
Я так и делаю, но все равно получаю ошибку, и ведь в details null. Сравнил даже какой получается хеш пароля при одном и том же salt - одинаковый. Я вероятно где то делаю простую ошибку, но не понимаю, что делаю неправильно

**Reply by lentryd** [reply]
> Я так и делаю, но все равно получаю ошибку, и ведь в details null. Сравнил даже какой получается хеш пароля при одном и том же salt - одинаковый. Я вероятно где то делаю простую ошибку, но не понимаю, что делаю неправильно

Если через либу на питоне все работает, то можешь попробовать сохранять абсолютно все куки или же проверь заголовки, которые передаешь. Если не заработает на питоне, то значит наши методы не подходят для твоего сайта.

**Reply by TheEntropyShard** [reply]
Сайт у меня тот же что и у вас, через питоновскую библиотеку работает все, попробую твои советы
